tests/test_rpm_verifier.py

```diff
@@ -355,24 +356,34 @@ def mock_image_processor(self, monkeypatch: MonkeyPatch) -> MagicMock:
         return mock
 
     @pytest.fixture()
-    def mock_generate_output(self, monkeypatch: MonkeyPatch) -> MagicMock:
-        """Monkey-patched generate_output"""
-        mock = create_autospec(generate_output, return_value=(False, ""))
-        monkeypatch.setattr(rpm_verifier, generate_output.__name__, mock)
-        return mock
+    def create_generate_output_mock(
+        self, monkeypatch: MonkeyPatch
+    ) -> Callable[[bool], MagicMock]:
+        """Create a generate_output mock with different results according to
+        the `fail_unsigned` flag"""
+
+        def _mock_generate_output(fail: bool = False) -> MagicMock:
+            """Monkey-patched generate_output"""
+            mock = create_autospec(generate_output, return_value=(fail, ""))
+            monkeypatch.setattr(rpm_verifier, generate_output.__name__, mock)
+            return mock
+
+        return _mock_generate_output
 
     def test_main(
         self,
-        mock_generate_output: MagicMock,
+        create_generate_output_mock: MagicMock,
         mock_image_processor: MagicMock,
     ) -> None:
         """Test call to rpm_verifier.py main function"""
```

Comment:
```suggestion
        generate_output_mock = create_generate_output_mock(fail=False)
        rpm_verifier.main(  # pylint: disable=no-value-for-parameter
            args=[
                "--input",
                "img1",
                "--fail-unsigned",
                sentinel.fail_on_unsigned_rpms,
                "--workdir",
                "some/path",
            ],
            obj={},
            standalone_mode=False,
        )
```

Reply:
Ignore the deleted part of the suggestion. I'm saying that `fail_on_unsigned_rpms` here has no affect.